mlkem/native/x86_64/src/basemul.S

```diff
@@ -115,23 +104,56 @@ vmovdqa		%ymm11,(64*\off+48)*2(%rdi)
 .text
 .global MLKEM_ASM_NAMESPACE(basemul_avx2)
 MLKEM_ASM_NAMESPACE(basemul_avx2):
-mov		%rsp,%r8
+mov		%rsp,%r11
 and		$-32,%rsp
 sub		$32,%rsp
 
-lea		(_ZETAS_EXP+176)*2(%rcx),%r9
 schoolbook	0
 
-add		$32*2,%r9
 schoolbook	1
 
-add		$192*2,%r9
 schoolbook	2
 
-add		$32*2,%r9
 schoolbook	3
 
-mov		%r8,%rsp
+mov		%r11,%rsp
+ret
+
+
+.macro mulcache_compute_iter i
+vmovdqa (64*\i+16)*2(%rsi), %ymm2
+vmovdqa (64*\i+48)*2(%rsi), %ymm3
+vmovdqa (16*\i)*2(%rdx), %ymm4
+
+vpmullw %ymm2, %ymm1, %ymm5
```

Comment:
Please store the twisted zetas as well, so you don't need to manually calculate the twists here.

Reply:
Fixed this